src/devices/Tlc59711/Tlc59711.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Iot.Device.Spi;
+
+namespace Iot.Device.Tlc59711
+{
+    /// <summary>
+    /// On-board LED on the device.
+    /// </summary>
+    public class Tlc59711 : IDisposable
+    {
+        private readonly ushort[] _pwmBuffer;
+        private readonly byte _numberOfDrivers;
+        private SpiDevice _spiDevice;
+        private byte _brightnessRed;
+        private byte _brightnessGreen;
+        private byte _brightnessBlue;
+        private byte[] _dataToWrite;
+
+        /// <summary>
+        /// Creates a new instance of the Tlc59711.
+        /// </summary>
+        public Tlc59711(byte numberOfDrivers, SpiDevice spiDevice)
+        {
+            _numberOfDrivers = numberOfDrivers;
```

Comment:
- can this be somehow read by SPI? if so might be worth doing it and making it optional.
- numberOfDrivers should be an `int`
- optional: for some reason I feel like arguments should be swapped but I'm not certain if we have any official guidelines on that

Reply:
No it can't be read, you can daisy-chain as many devices as you like, but there's no reading anything from the device, the device is input-only (why I started down the path of output-only SPI device).
I can change the parameter to be int, just figured it could never be that large.
I can swap parameters and default drivers to 1 (most common)